code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -28,10 +27,11 @@
 	opaque = FALSE
 	alpha = 100
 
-
 /obj/machinery/smoke_machine/Initialize()
 	. = ..()
-	create_reagents(volume)
+	create_reagents(100)
+	for(var/obj/item/stock_parts/matter_bin/B in component_parts)
+		reagents.maximum_volume += 100 * B.rating
```

Comment:
keep the volume var and when you calculate based on parts do `initial(volume) * B.rating` so in the future only the one place has to be edited to buff or nerf

Reply:
I'll try another approach, see if it is ok.